File: SDKs/cocos2dx/Classes/IntelliVerseX/IVXDiscordModeration.cpp
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

#include "IntelliVerseX/IVXDiscordModeration.h"
#include <stdexcept>

namespace IntelliVerseX {

IVXDiscordModeration& IVXDiscordModeration::getInstance() {
    static IVXDiscordModeration instance;
    return instance;
}

void IVXDiscordModeration::enableAutoModeration(bool) {
    throw std::runtime_error("Not implemented");
}

void IVXDiscordModeration::processModerationMetadata(const std::string&, const std::map<std::string, std::string>&) {
    throw std::runtime_error("Not implemented");
}

IVXModerationDecision IVXDiscordModeration::getModerationAction(const std::map<std::string, std::string>&) {
    throw std::runtime_error("Not implemented");
}

void IVXDiscordModeration::startVoiceModerationCapture(const std::string&) {
    throw std::runtime_error("Not implemented");
}

void IVXDiscordModeration::stopVoiceModerationCapture() {
    throw std::runtime_error("Not implemented");
}

void IVXDiscordModeration::reportUser(const std::string&, const std::string&, std::function<void(bool)> onComplete) {
    if (onComplete) onComplete(false);
}

} // namespace IntelliVerseX

File: SDKs/java/src/main/java/com/intelliversex/sdk/discord/IVXDiscordModeration.java
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

package com.intelliversex.sdk.discord;

import java.util.Map;
import java.util.concurrent.CompletableFuture;

/**
 * Discord moderation metadata, voice capture for moderation, user reporting.
 * Stub API matching Unity {@code IVXDiscordModeration}.
 */
public final class IVXDiscordModeration {

    private static final IVXDiscordModeration INSTANCE = new IVXDiscordModeration();

    private volatile boolean autoModerateEnabled = true;

    private IVXDiscordModeration() {}

    public static IVXDiscordModeration getInstance() {
        return INSTANCE;
    }

    public boolean isAutoModerateEnabled() {
        return autoModerateEnabled;
    }

    public void setAutoModerateEnabled(boolean autoModerateEnabled) {
        this.autoModerateEnabled = autoModerateEnabled;
    }

    public void enableAutoModeration(boolean enable) {
        throw new UnsupportedOperationException("Not implemented");
    }

    public void processModerationMetadata(String messageId, Map<String, String> metadata) {
        throw new UnsupportedOperationException("Not implemented");
    }

    public static IVXModerationDecision getModerationAction(Map<String, String> metadata) {
        throw new UnsupportedOperationException("Not implemented");
    }

    public void startVoiceModerationCapture(String lobbyId) {
        throw new UnsupportedOperationException("Not implemented");
    }

    public void stopVoiceModerationCapture() {
        throw new UnsupportedOperationException("Not implemented");
    }

    public CompletableFuture<Boolean> reportUser(String userId, String reason) {
        CompletableFuture<Boolean> f = new CompletableFuture<>();
        f.completeExceptionally(new UnsupportedOperationException("Not implemented"));
        return f;
    }

    public enum IVXModerationAction {
        Show,
        Hide,
        Blur,
        Replace
    }

    public static final class IVXModerationDecision {
        public String messageId;
        public IVXModerationAction action;
        public String reason;
        public String replacement;
        public String severity;
    }
}
